pkg/source/reconciler/mtsource/controller.go

```diff
@@ -73,7 +81,53 @@ func NewController(
 	sourcesv1beta1.RegisterAlternateKafkaConditionSet(sourcesv1beta1.KafkaMTSourceCondSet)
 
 	rp := time.Duration(env.SchedulerRefreshPeriod) * time.Second
-	c.scheduler = scheduler.NewScheduler(ctx, system.Namespace(), mtadapterName, c.vpodLister, rp, env.PodCapacity, env.SchedulerPolicy, nodeInformer.Lister())
+	evictor := func(vpod scheduler.VPod, from *duckv1alpha1.Placement) error {
+		key := vpod.GetKey()
+		sources := kafkaclient.Get(ctx).SourcesV1beta1().KafkaSources(key.Namespace)
+
+		before, err := sources.Get(ctx, key.Name, metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+
+		after := before.DeepCopy()
+
+		bp := after.GetPlacements()
+		ap := make([]duckv1alpha1.Placement, 0, len(bp)-1)
+		for _, p := range bp {
+			if p.PodName != from.PodName {
+				ap = append(ap, p)
+			}
+		}
+		after.Status.Placement = ap
```

Comment:
1. If we have a Kafka source on the last pod with one replica and this Kafka source has only one vreplica. What would happen? 
2. Once we remove the replica from the last pod, what if KEDA scaled the replica again? Almost seems like KEDA and this logic works in the opposite direction.

Reply:
1. if there are enough free capacity before the last pod, the vreplica will be moved
2. Keda is about increasing/decreasing the number of vreplicas. But I see what you mean. When scaling down, the vreplicas closest to the last pod are first removed (after your fix) so from this PoV Keda and this PR are working in the same direction.